tail/tail_test.go

```diff
@@ -85,6 +85,19 @@ func TestTailFuzz(t *testing.T) {
 			t.Fatalf("record %d doesn't match", i)
 		}
 	}
+	// Attempt to read one more record, but expect no more records.
+	// Give the reader a chance to run for a while, then cancel its
+	// context so the test doesn't time out.
+	go func() {
+		time.Sleep(time.Second)
+		cancel()
+	}()
+	if wr.Next() {
```

Comment:
if the previous condition `wr.Next()` returns `false` for loop in line 74, this wr.Next() must not run based on the description of [wal.Reader.Next()](https://github.com/Stackdriver/stackdriver-prometheus-sidecar/blob/abebfbf954b702a77bfc73f4c833298e38e849ad/vendor/github.com/prometheus/tsdb/wal/wal.go#L685-L687).

Reply:
True. But if `wr.Next()` had returned `false`, the test should have aborted with a Fatal error above where the length of the read and write arrays are compared, right?

Reply:
If wr.Next() in line 74 had returned `false` because internal read returns `io.EOF` ([code](https://github.com/Stackdriver/stackdriver-prometheus-sidecar/blob/abebfbf954b702a77bfc73f4c833298e38e849ad/vendor/github.com/prometheus/tsdb/wal/wal.go#L689-L691)), `wr.Err()` in line 77 would return `nil`, so it would not abort.

Reply:
Wouldn't it abort on line 80?

Reply:
PTAL.

Reply:
LGTM.